plugins/pulp_rpm/plugins/db/models.py

```diff
@@ -1179,22 +1179,64 @@ def _check_packages(existing_packages, new_packages):
             return existing_packages == new_packages
         return False
 
+    def create_legacy_metadata_dict(self):
+        """
+        Generate metadata dict and add erratum pkglist to it since it is stored
+        in a separate MongoDB collection.
+
+        This dict is used during incremetal export of erratum in PublishErrataStepIncremental.
+        """
+        metadata_dict = super(Errata, self).create_legacy_metadata_dict()
+        pkglists = models.ErratumPkglist.objects(errata_id=erratum_unit.errata_id)
+        metadata_dict['pkglist'] = [pkglist['collections'] for pkglist in pkglists]
+        return metadata_dict
+
+    def import_erratum(self, repo_id):
+        """
+        To avoid importing erratum units without pkglist:
+        - merge and save pkglist to its own collection
+        - then save erratum unit itself
```

Comment:
They will be cleaned up during orphan removal: first erratum itself and then pkglists. If something goes wrong between these two deletions, it is possible that pkglists won't be cleaned up.
Removing just any pkglists which refer to a non-existing erratum is not safe, I think, because of the race condition between orphan cleanup and sync. And because erratum unit will be separated into two pieces and stored in two collections, the chances for race condition are higher than for usual one-collection units.